### PR TITLE
Issue 1461

### PR DESCRIPTION
**Pull request by koheiw**
I fully inspected the code for #1461, and made changes where improvement is possible.

### CODE REVIEW

**Comment by koheiw** [comment]
I did not change the tests, but I am not sure about the diagonal counts.

For example, I think A-A should be 4 instead of 2

https://github.com/quanteda/quanteda/blob/8b7d2235fb13a10ecd6ebf40cdf778e67fa77e0d/tests/testthat/test-fcm.R#L34-L45

**Comment by koheiw** [comment]
I removed `span_sentence` because it was not dong anything. We can add this provide this functionality but, in principle, users should segment documents in to sentences beforehand.

File: tests/testthat/test-fcm.R [hunk 1]
```diff
@@ -1,31 +1,42 @@
 context('Testing fcm*.R')
 
-test_that("compare the output feature co-occurrence matrix to that of the text2vec package", {
-    skip_if_not_installed("text2vec")
-    library("text2vec")
-    
+# test_that("compare the output feature co-occurrence matrix to that of the text2vec package", {
```
Comment (kbenoit) [inline tests/testthat/test-fcm.R]: Our diagonals are different but does it make sense to remove this comparison? (They match, except the diagonal.)

If we do remove this code then we can remove **text2vec** from `Suggests:`.

**Comment by kbenoit** [comment]
Moving a comment here from another issue where I pasted it by mistake! Just for the record, since I am happy with the removal of this argument.

That's true, I had forgotten we did not implement `span_sentence = FALSE`. I think the idea was to make it easier than this:

``` r
library("quanteda")
## Package version: 1.4.3
txt <- "A b c d.  A a b."

fcm(txt, context = "window", size = 3, span_sentence = FALSE)
## Warning in fcm.tokens(tokens(x), ...): spanSentence = FALSE not yet
## implemented

corpus(txt) %>%
  corpus_reshape(to = "sentences") %>%
  tokens(remove_punct = TRUE) %>%
  tokens_tolower() %>%
  fcm(context = "window", size = 3)
## Feature co-occurrence matrix of: 4 by 4 features.
## 4 x 4 sparse Matrix of class "fcm"
##         features
## features a b c d
##        a 1 3 1 1
##        b 0 0 1 1
##        c 0 0 0 1
##        d 0 0 0 0

# compare to:
tokens(txt, remove_punct = TRUE) %>%
  tokens_tolower() %>%
  fcm(context = "window", size = 3, span_sentence = TRUE)
## Feature co-occurrence matrix of: 4 by 4 features.
## 4 x 4 sparse Matrix of class "fcm"
##         features
## features a b c d
##        a 3 5 3 3
##        b 0 1 2 2
##        c 0 0 0 1
##        d 0 0 0 0

```



But I agree this ought to be a pre-processing step, separate from `fcm()`.

The lack of a `tolower` option in `fcm()` suggests we should add one, or consider removing it from `dfm()`. Maybe it's time to consider pruning some of the shortcut options in `dfm()`...

**Pull request merged**